refactor build_deploy.sh

Victoremepunto opened the pull request:
This changes provides the following:
- Reverts multiarch support 
- Improved handling of commands evaluation
- refactor to use [CICD tools repo](https://github.com/RedHatInsights/cicd-tools) which provides:
    - Add temporary labels when building if PR context
    - Agnostic container engine commands
    - Automatic container registry login
    - Local runs enabled to avoid `push` commands if running locally (outside of a CI job)

Victoremepunto commented:
/retest

Victoremepunto commented:
/retest

adamrdrew commented:
/retest

adamrdrew reviewed (approved):
This looks good to me! You'll notice I didn't put any comments in the main script changes. That's because I ran it locally and it worked fine.  I looked at it line by line and it all seems sane and readable. I will say I had to ask chatgpt to tell me what these two lines did

```
[[ 1 -eq $(jq '.tags | length' <<<"$response") ]]
```

```
! git diff --quiet "$target_branch" -- "${BASE_IMAGE_FILES[@]}"
```

If you felt moved to throw a comment above each those lines I think it would help readability a bit. But great patch thanks for doing this!

Victoremepunto commented:
> This looks good to me! You'll notice I didn't put any comments in the main script changes. That's because I ran it locally and it worked fine. I looked at it line by line and it all seems sane and readable. I will say I had to ask chatgpt to tell me what these two lines did
> 
> ```
> [[ 1 -eq $(jq '.tags | length' <<<"$response") ]]
> ```
> 
> ```
> ! git diff --quiet "$target_branch" -- "${BASE_IMAGE_FILES[@]}"
> ```
> 
> If you felt moved to throw a comment above each those lines I think it would help readability a bit. But great patch thanks for doing this!

Hey Adam, thanks for the review.

On the first one, I didn't write that line, and it already had a comment, which I left intact. it reads...

```
 33   # find all non-expired tags
 34   [[ 1 -eq $(jq '.tags | length' <<<"$response") ]]
```

The second one, I've added a comment but just because you asked nicely :P ...  sorry, now joking aside, I was hoping the function name would provide enough context. But I added the comment anyways, the function looks like this now:

```
 46 _base_image_files_changed() {
 47 
 48   local target_branch=${ghprbTargetBranch:-master}
 49 
 50   # Use git to check for any non staged differences in the Base Image files
 51   ! git diff --quiet "$target_branch" -- "${BASE_IMAGE_FILES[@]}"
 52 }
```

Hope that helps !

Victoremepunto commented on Dockerfile:
```diff
@@ -15,7 +15,7 @@ COPY controllers/ controllers/
 
 # Build
-RUN CGO_ENABLED=1 GOOS=linux GO111MODULE=on go build -o manager main.go
+RUN CGO_ENABLED=1 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o manager main.go
```
done!

Victoremepunto commented on Makefile:
```diff
@@ -134,7 +134,7 @@ docker-build: update-version
 # Build the docker image
 docker-build-no-test-quick: update-version
-	CGO_ENABLED=0 GOOS=linux GO111MODULE=on go build -o bin/manager-cgo main.go
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -o bin/manager-cgo main.go
```
OK
